src/clickonce/native/projects/NetCoreCheck/CA/NetCoreCheckCA.cpp

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "NetCoreCheckCA.h"
+#include "MsiLogger.h"
+
+#define CHECKNETRUNTIME_PARAM_FRAMEWORK_PROPERTY_NAME L"CheckNETRuntime_Framework"
+#define CHECKNETRUNTIME_PARAM_VERSION_PROPERTY_NAME   L"CheckNETRuntime_Version"
+#define CHECKNETRUNTIME_RESULT_PROPERTY_NAME          L"CheckNETRuntime_Result"
+
+#define ExitOnFailure(x, s, ...)   if (FAILED(x)) { msiWrapper.LogFailure(x, s, __VA_ARGS__);  goto Exit; }
+#define FreeStr(s) if (s) { delete s; }
+
+// Globals
+Logger *g_log;
+
+UINT __stdcall CheckNETRuntime(MSIHANDLE hInstall)
+{
+    int ret = EXIT_SUCCESS;
+    LPWSTR frameworkName = NULL;
+    LPWSTR frameworkVersion = NULL;
+    MsiWrapper msiWrapper(hInstall);
+
+    MsiLogger logger;
+    logger.Initialize(&msiWrapper);
+    g_log = &logger;
```

Comment:
Good question, I wasn't able to find an answer with a few minutes of searching.  But in this case it doesn't matter since MsiWrapper and by extension MsiLogger are just thin wrappers around an MSIHANDLE that will be effectively the same even in the unlikely case that this custom action is run twice during a single install.  There's no initialization/cleanup that we need to worry about happening too many/few times.  Still, when looking at this I realized that having an "Initialize" method for this was counter-intuitive, so I'm now just passing the handle to the constructor.

Reply:
I got curious and realized I could try this pretty easily, and I found that at the beginning of the second execution g_log was indeed NULL.